scripts/quickstart

```diff
@@ -1,94 +1,145 @@
-  echo "  -d DOMAIN    the domain name this deployment will run as, eg. example.com. Default is 'openbalena.local'"
-  echo "  -U EMAIL     the email address of the superuser account, used to login to your install from the Balena CLI"
-  echo "  -P PASSWORD  the password to use for the superuser account."
-  echo
-}
+BLACK=`tput setaf 0`
+RED=`tput setaf 1`
+GREEN=`tput setaf 2`
+YELLOW=`tput setaf 3`
+BLUE=`tput setaf 4`
+MAGENTA=`tput setaf 5`
+CYAN=`tput setaf 6`
+WHITE=`tput setaf 7`
+
+BOLD=`tput bold`
+RESET=`tput sgr0`
+
+quickstart() {
+
+  # for macos machines, we need proper OpenSSL...
+  if [[ "$OSTYPE" == "darwin"* ]]; then
+    echo -e "${BLUE}INFO: macOS detected. Bringing in OpenSSL if installed...${RESET}"
+    export PATH="/usr/local/opt/openssl/bin:$PATH"
+  fi
+
+  OPENSSL_VERSION=$(openssl version -v)
+  if [[ "${OPENSSL_VERSION}" =~ ^LibreSSL.*$ ]]; then
+    echo -e "${RED}ERROR: You may not have a compatible OpenSSL version (${OPENSSL_VERSION}). Please install OpenSSL version 1.0.2q or above.${RESET}"
+    exit 1
+  fi
+
+  REALPATH=
+  REALPATHS=(
+    'realpath'
+    'readlink'
+    'greadlink -f'
+  )
+  for cmd in "${REALPATHS[@]}"
+  do
+    if type ${cmd%% *} &>/dev/null; then
+      REALPATH=$cmd
+      break
+    fi
+  done
+
+  if [ -z $REALPATH ]; then
+    echo "${RED}ERROR: Unable to find suitable command for realpath.${RESET}"
+    exit 1
+  fi
+  
+  realpath() {
+    echo $(command $REALPATH "$@")
+  }
+
+  CMD="$(realpath "$0")"
+  DIR="$(dirname "${CMD}")"
+  BASE_DIR="$(dirname "${DIR}")"
+  CONFIG_DIR="${BASE_DIR}/config"
+  CERTS_DIR="${CONFIG_DIR}/certs"
+
+  DOMAIN=openbalena.local
+
+  usage() {
+    echo "usage: $0 [-h] [-p] [-d DOMAIN] -U EMAIL -P PASSWORD"
+    echo
+    echo "  -p           patch hosts - patch the host /etc/hosts file"
+    echo "  -d DOMAIN    the domain name this deployment will run as, eg. example.com. Default is 'openbalena.local'"
+    echo "  -U EMAIL     the email address of the superuser account, used to login to your install from the Balena CLI"
+    echo "  -P PASSWORD  the password to use for the superuser account."
+    echo
+  }
+
+  show_help=false
+  patch_hosts=false
+  while getopts ":hpxd:U:P:" opt; do
+    case "${opt}" in
+      h) show_help=true;;
+      p) patch_hosts=true;;
+      x) set -x;;
+      d) DOMAIN="${OPTARG}";;
+      U) SUPERUSER_EMAIL="${OPTARG}";;
+      P) SUPERUSER_PASSWORD="${OPTARG}";;
+      *)
+        echo "Invalid argument: -${OPTARG}"
+        usage
+        exit 1
+        ;;
+    esac
+  done
+  shift $((OPTIND-1))
+
+  if [ -z "${SUPERUSER_EMAIL}" ] || [ -z "${SUPERUSER_PASSWORD}" ]; then
+    usage
+    exit 1
+  fi
+
+  if [ "$show_help" = "true" ]; then
+    usage
+    exit 1
+  fi
+
+  echo_bold() {
+    printf "\\033[1m%s\\033[0m\\n" "${@}"
+  }
+
+  echo_bold "==> Creating new configuration at: $CONFIG_DIR"
+  mkdir -p "$CONFIG_DIR" "$CERTS_DIR"
+
+  source "${DIR}/ssl-common.sh"
```

Comment:
I don't think this is necessary.

Reply:
Hmm, it might make the _output_ cleaner but I'd say it certainly makes the code "less clean" (dirtier?).

Reply:
I think I'd rather we did something like

```bash
echo_bold '==> Bootstrapping easyrsa...'
source "${DIR}/ssl-common.sh"
```

and drop the echos from `ssl-common.sh`. 

That way the code and the logs read sensibly, not one or the other.